src/Hydra/JsonSchema/SchemaFactory.php

```diff
@@ -63,10 +62,7 @@ final class SchemaFactory implements SchemaFactoryInterface
     public function __construct(SchemaFactoryInterface $schemaFactory)
     {
         $this->schemaFactory = $schemaFactory;
-
-        if ($schemaFactory instanceof BaseSchemaFactory) {
```

Comment:
Why not keeping the check?

Reply:
I've moved the check now to the public method addDistinctFormat from the hydra schema factory.